src/core/elevation/qgsabstractprofilegenerator.h

```diff
@@ -157,6 +157,37 @@ class CORE_EXPORT QgsAbstractProfileResults
      */
     virtual QgsDoubleRange zRange() const = 0;
 
+    /**
+     * \brief Encapsulates results of snapping a point.
+     * \ingroup core
+     * \since QGIS 3.26
+     */
+    class SnapResult
+    {
+      public:
+
+        //! Snapped point distance along curve
+        double snappedDistanceAlongCurve = std::numeric_limits< double >::quiet_NaN();
+
+        //! Snapped point height
+        double snappedHeight = std::numeric_limits< double >::quiet_NaN();
+
+        //! Distance snapped
+        double snapDistance = std::numeric_limits< double >::max();
```

Comment:
I'll disambiguate, but yes -- it's supposed to represent the "distance moved during the snap", so that we can pick the match from different sources which is closest